bindings/rust/build.rs

```diff
@@ -1,170 +1,58 @@
-#[cfg(feature = "build_unicorn_cmake")]
-use bytes::Buf;
-#[cfg(feature = "build_unicorn_cmake")]
-use flate2::read::GzDecoder;
 #[cfg(feature = "use_system_unicorn")]
 use pkg_config;
 #[cfg(feature = "build_unicorn_cmake")]
-use reqwest::header::USER_AGENT;
-#[cfg(feature = "build_unicorn_cmake")]
-use std::path::{Path, PathBuf};
+use std::env;
 #[cfg(feature = "build_unicorn_cmake")]
-use std::{env, process::Command};
-#[cfg(feature = "build_unicorn_cmake")]
-use tar::Archive;
-
-#[cfg(feature = "build_unicorn_cmake")]
-fn find_unicorn(unicorn_dir: &Path) -> Option<PathBuf> {
-    for entry in std::fs::read_dir(unicorn_dir).ok()? {
-        let entry = entry.unwrap();
-        let path = entry.path();
-
-        if path.is_dir() && path.file_name()?.to_str()?.contains("unicorn") {
-            return Some(path);
-        }
+use std::path::Path;
+
+#[cfg(all(feature = "build_unicorn_cmake", target_os = "windows"))]
+fn setup_env_windows() {
+    let devenv_path = cc::windows_registry::find_tool(
+        env::var("TARGET").unwrap().as_str(), "devenv").unwrap();
+    let devenv_dir = devenv_path.path().parent().unwrap();
+
+    let cmake_path = devenv_dir.join(r"CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe");
```

Comment:
This seems not good.

I feel it better to call `cmake` directly, which also works for msys2 if it (really) works.

Reply:
This is the best way for msvc. I'm not sure about msys2 (i assume you mean mingw).

Reply:
https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-170#use-the-developer-tools-in-an-existing-command-window

Reply:
this crate is used by other rust projects; i.e. i just want to type `cargo run` and it works. similarly, other tools (like rust language servers) will break if the build.rs doesn't "just work".

Reply:
> this crate is used by other rust projects; i.e. i just want to type `cargo run` and it works. similarly, other tools (like rust language servers) will break if the build.rs doesn't "just work".

I mean, if we really would like to help users build it without the VS prompt, we shouldn't do such hack which might break in a further MSVC release.

Reply:
Anything can break on future msvc/VS release. do you see the big red warning directly above the section you linked? :)

Reply:
But my concern is that: Could `CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe` change?

Reply:
It's unlikely, but it could. It's equally likely the location of vsvars script itself changes. The actual solution is pushing the problem to be handled by the `cc` (or maybe `cmake`) crate, but for now they don't have the functionality. In real life, everyone has to update their tools with each new VS, anyway (it's just that it should be `cc` crate's problem to deal with that, not unicorn's).